c2rust-transpile/src/translator/variadic.rs

```diff
@@ -9,130 +9,110 @@ pub enum VaPart {
+            CExprKind::DeclRef(_, va_id, _) => va_id }?;
             Some(va_id)
-        }
+        };
 
-        match_vastart_struct(&self.ast_context, expr)
-            .or_else(|| match_vastart_pointer(&self.ast_context, expr))
-            .or_else(|| match_vastart_struct_member(&self.ast_context, expr))
-            .or_else(|| match_vastart_struct_pointer_member(&self.ast_context, expr))
+        match_vastart_struct()
+            .or_else(match_vastart_pointer)
+            .or_else(match_vastart_struct_member)
+            .or_else(match_vastart_struct_pointer_member)
     }
 
     pub fn match_vaend(&self, expr: CExprId) -> Option<CDeclId> {
         self.match_vastart(expr)
     }
 
     pub fn match_vacopy(&self, dst_expr: CExprId, src_expr: CExprId) -> Option<(CDeclId, CDeclId)> {
-        let dst_id = self.match_vastart(dst_expr);
-        let src_id = self.match_vastart(src_expr);
-        if let (Some(did), Some(sid)) = (dst_id, src_id) {
-            return Some((did, sid));
-        }
-        None
+        let dst_id = self.match_vastart(dst_expr)?;
+        let src_id = self.match_vastart(src_expr)?;
+        Some((dst_id, src_id))
     }
 
     pub fn match_vapart(&self, expr: CExprId) -> Option<VaPart> {
-        match_or! { [self.ast_context[expr].kind]
-        CExprKind::Call(_, func, ref args) => (func, args) }
-        match_or! { [self.ast_context[func].kind]
-        CExprKind::ImplicitCast(_, fexp, CastKind::BuiltinFnToFnPtr, _, _) => fexp }
-        match_or! { [self.ast_context[fexp].kind]
-        CExprKind::DeclRef(_, decl_id, _) => decl_id }
-        match_or! { [self.ast_context[decl_id].kind]
-        CDeclKind::Function { ref name, .. } => name }
-        match name as &str {
-            "__builtin_va_start" => {
-                if args.len() != 2 {
-                    return None;
-                }
-                self.match_vastart(args[0]).map(VaPart::Start)
-            }
-
-            "__builtin_va_copy" => {
-                if args.len() != 2 {
-                    return None;
-                }
-                self.match_vacopy(args[0], args[1])
-                    .map(|(did, sid)| VaPart::Copy(did, sid))
-            }
-
-            "__builtin_va_end" => {
-                if args.len() != 1 {
-                    return None;
-                }
-                self.match_vaend(args[0]).map(VaPart::End)
-            }
-
+        let (func, args) = match_or! { [self.ast_context[expr].kind]
+        CExprKind::Call(_, func, ref args) => (func, args) }?;
+        let fexp = match_or! { [self.ast_context[func].kind]
+        CExprKind::ImplicitCast(_, fexp, CastKind::BuiltinFnToFnPtr, _, _) => fexp }?;
+        let decl_id = match_or! { [self.ast_context[fexp].kind]
+        CExprKind::DeclRef(_, decl_id, _) => decl_id }?;
+        let name = match_or! { [self.ast_context[decl_id].kind]
+        CDeclKind::Function { ref name, .. } => name }?;
```

Comment:
Indent the line continuations within the `match_or` macro invocations here to make it clearer that they are not their own expressions at the same nesting level as the `let`s.

Reply:
I'm pretty sure I tried that but `cargo fmt` changed it back to this.  Let me check, though.  The macro would have to be re-written to work better with `rustfmt`.

Reply:
I can't indent them according to `cargo fmt`, but I can make each `match_or!` on 1 line only.

Reply:
Would this look better?

```rust
        let (func, args) = match_or! { [ast[expr].kind] CExprKind::Call(_, func, ref args) => (func, args) }?;
        let fexp = match_or! { [ast[func].kind] CExprKind::ImplicitCast(_, fexp, CastKind::BuiltinFnToFnPtr, _, _) => fexp }?;
        let decl_id = match_or! { [ast[fexp].kind] CExprKind::DeclRef(_, decl_id, _) => decl_id }?;
        let name = match_or! { [ast[decl_id].kind] CDeclKind::Function { ref name, .. } => name }?;
```

Reply:
Fixed in https://github.com/immunant/c2rust/pull/638/commits/b31901c159df78f5142e2a02430527f946025b81 assuming that looks good.